lib/utils.js

```diff
@@ -39,8 +39,31 @@ function containsModifier(parameter, modifier) {
   });
 }
 
+/*
+ *  Returns the value when the keys contain complex strings that cannot be indexed with a dot notation
+ */
+function getArrayPath(obj, path) {
+  let fullPath = [];
+  console.log(path)
+  for (let key of Object.keys(obj)) {
+    if (path === key) {
+      // We are the tail of the tree
+      fullPath.push(key);
+    } else {
+      const subKeys=path.split(`${key}.`);
+      if (subKeys.length > 1 && path.startsWith(`${key}.`)) {
```

Comment:
Not sure if I'm understanding this right. What would be the behavior of this if, for example, we encountered the following corner case:
```
  ingress:
    hostname:
      - "hostname.test"
```

Reply:
that should not be a corner case as what has a dot there is the value not the key. Could you clarify the question?

Reply:
You are right, I double-checked this and it would work as expected.